jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/editor/CloudWatchLogStream.kt

```diff
@@ -33,107 +24,94 @@ import software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions.TailLogs
 
     init {
-        logStreamActor = LogStreamActor(project.awsClient(), logsTable, logGroup, logStream)
-        Disposer.register(this, logStreamActor)
+        logsPanel.setContent(logStreamTable.component)
+        Disposer.register(this, logStreamTable)
         searchLabel.text = "${project.activeCredentialProvider().displayName} => ${project.activeRegion().displayName} => $logGroup => $logStream"
         searchField.emptyText.text = message("cloudwatch.logs.filter_logs")
-        logsTable.autoResizeMode = JTable.AUTO_RESIZE_ALL_COLUMNS
-        logsPanel.verticalScrollBar.addAdjustmentListener {
-            if (logsTable.model.rowCount == 0) {
-                return@addAdjustmentListener
-            }
-            if (logsPanel.verticalScrollBar.isAtBottom()) {
-                launch {
-                    // Don't load more if there is a logStreamingJob because then it will just keep loading forever at the bottom
-                    if (logStreamingJob == null) {
-                        logStreamActor.channel.send(LogStreamActor.Messages.LOAD_FORWARD)
-                    }
-                }
-            } else if (logsPanel.verticalScrollBar.isAtTop()) {
-                launch { logStreamActor.channel.send(LogStreamActor.Messages.LOAD_BACKWARD) }
+
+        addAction()
+        addActionToolbar()
+        addSearchListener()
+
+        launch {
+            if (startTime != null && duration != null) {
+                logStreamTable.channel.send(LogStreamActor.Message.LOAD_INITIAL_RANGE(startTime, duration))
+            } else {
+                logStreamTable.channel.send(LogStreamActor.Message.LOAD_INITIAL())
             }
         }
-        launch {
-            try {
-                if (startTime != null && duration != null) {
-                    logStreamActor.loadInitialRange(startTime, duration)
+    }
+
+    private fun addSearchListener() {
+        searchField.addActionListener(object : ActionListener {
+            private var lastText = ""
+            override fun actionPerformed(e: ActionEvent?) {
+                val searchFieldText = searchField.text.trim()
+                if (searchFieldText == lastText) {
+                    return
+                }
+                lastText = searchFieldText
+                val oldTable = searchStreamTable
+                // If it is empty, replace the table with the original table
+                if (searchFieldText.isEmpty()) {
+                    searchStreamTable = null
+                    launch(edtContext) {
+                        logsPanel.setContent(logStreamTable.component)
+                        // Dispose the old one if it was not null
+                        oldTable?.let { launch { Disposer.dispose(it) } }
+                    }
                 } else {
-                    logStreamActor.loadInitial()
+                    // This is thread safe because the actionPerformed is run on the UI thread
```

Comment:
maybe add an assert then?

Reply:
This is done in an action listener and it can't be referenced anywhere so I don't think we need one